packages/squiggle-lang/src/rescript/Distributions/PointSetDist/PointSetDist_Scoring.res

```diff
@@ -1,46 +1,185 @@
+            ~combineFn,
+            ~integrateFn,
+            ~toMixedFn,
+          ),
+        )->E.R2.fmap(((discretePart, continuousPart)) => discretePart +. continuousPart)
+      | (_, _, _, _) => `unreachable state`->Operation.Other->Error
+      }
     }
+
+  let sumWithPrior = (
+    ~estimate: t,
+    ~answer: t,
+    ~prior: t,
+    ~combineFn,
+    ~integrateFn,
+    ~toMixedFn,
+  ): result<float, Operation.Error.t> => {
+    let kl1 = sum(~estimate, ~answer, ~combineFn, ~integrateFn, ~toMixedFn)
+    let kl2 = sum(~estimate=prior, ~answer, ~combineFn, ~integrateFn, ~toMixedFn)
+    E.R.merge(kl1, kl2)->E.R2.fmap(((kl1', kl2')) => kl1' -. kl2')
+  }
 }
 
-module LogScoreWithPointResolution = {
-  let logFn = Js.Math.log
-  let score = (
-    ~priorPdf: option<float => float>,
-    ~predictionPdf: float => float,
-    ~answer: float,
+module WithScalarAnswer = {
+  let score' = (~estimatePdf: float => float, ~answer: float): result<float, Operation.Error.t> => {
+    let density = answer->estimatePdf
+    if density < 0.0 {
+      Operation.PdfInvalidError->Error
+    } else if density == 0.0 {
+      infinity->Ok
+    } else {
+      density->logFn->(x => -.x)->Ok
+    }
+  }
+  let scoreWithPrior' = (
+    ~estimatePdf: float => float,
+    ~answer: scalar,
+    ~priorPdf: float => float,
   ): result<float, Operation.Error.t> => {
-    let numerator = answer->predictionPdf
-    if numerator < 0.0 {
+    let numerator = answer->estimatePdf
+    let priorDensityOfAnswer = answer->priorPdf
+    if numerator < 0.0 || priorDensityOfAnswer < 0.0 {
       Operation.PdfInvalidError->Error
-    } else if numerator == 0.0 {
+    } else if numerator == 0.0 || priorDensityOfAnswer == 0.0 {
       infinity->Ok
     } else {
-      -.(
-        switch priorPdf {
-        | None => numerator->logFn
-        | Some(f) => {
-            let priorDensityOfAnswer = f(answer)
-            if priorDensityOfAnswer == 0.0 {
-              neg_infinity
-            } else {
-              (numerator /. priorDensityOfAnswer)->logFn
-            }
-          }
-        }
-      )->Ok
+      minusScaledLogOfQuot(~esti=numerator, ~answ=priorDensityOfAnswer)
     }
   }
+
+  let sum = (mp: PointSetTypes.MixedPoint.t): float => mp.continuous +. mp.discrete
+  let score = (~estimate: t, ~answer: scalar): result<float, Operation.Error.t> => {
+    let estimatePdf = x =>
+      switch estimate {
+      | Continuous(esti) => Continuous.T.xToY(x, esti)->sum
+      | Discrete(esti) => Discrete.T.xToY(x, esti)->sum
+      | Mixed(esti) => Mixed.T.xToY(x, esti)->sum
+      }
+
+    score'(~estimatePdf, ~answer)
+  }
+  let scoreWithPrior = (~estimate: t, ~answer: scalar, ~prior: t): result<
+    float,
+    Operation.Error.t,
+  > => {
+    let estimatePdf = x =>
+      switch estimate {
+      | Continuous(esti) => Continuous.T.xToY(x, esti)->sum
+      | Discrete(esti) => Discrete.T.xToY(x, esti)->sum
+      | Mixed(esti) => Mixed.T.xToY(x, esti)->sum
```

Comment:
This seems pretty wrong to me. Discrete values and continuous values are completely different.

Reply:
My guess is that we should just throw an error at this point. We can treat it better later, in a different PR.